src/doc/CMakeLists.txt

```diff
@@ -0,0 +1,245 @@
+set(GEN_TEXI_FILES_LIST
+	version.texi
+	parts.texi
+	programmers.texi
+	programmer_types.texi
+	)
+
+set(DOC_TYPES_LIST
+	pdf
+	dvi
+	ps
+	html
+	)
+
+if("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
+	set(SCRIPT_DEBUG "TRUE")
+endif()
+
+# =====================================
+# Definitions
+# =====================================
+
+set(GEN_TEXI_SCRIPT "gen_texi.sh")
+set(GEN_DOC_SCRIPT "gen_doc.sh")
+set(GEN_VERSION_SCRIPT "gen_version.sh")
+set(AWK_SCRIPT "texify.awk")
+
+set(DOC_NAME ${PROJECT_NAME})
+
+set(MAIN_TEXI "${DOC_NAME}.texi")
+set(STYLE_SHEET "${DOC_NAME}.css")
+set(COMMENTS "parts_comments.txt")
+
+set(PLATFORM "${CMAKE_SYSTEM_NAME}")
+if (${MINGW})
+	set(PLATFORM "MINGW")
+endif()
+
+set(DOC_FILES_LIST "")
+set(TAR_BALL "${PROJECT_NAME}-doc.tar.gz")
+set(TAR_BALL_PREFIX_DIR "doc")
+
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
+	if(${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "arm64")
+		set(BREW_PREFIX "/opt/homebrew/opt")
```

Comment:
@dl8dtl You mean this might be `/opt/local/` for `texinfo` as well on M1?
Or do you mean by macports some other way of installing packages?
Sadly I have no clue how to install packages on mac. That brew stuff I figured out with the help of @mcuee and the X86 CI logs.

Reply:
macports is similar to homebrew and it is also popular. You can refer to the following.
https://github.com/avrdudes/avrdude/blob/main/build.sh#L46